File: logistics_v2/checkout_receipt.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime
from decimal import Decimal


@dataclass(frozen=True, slots=True)
class CheckoutReceipt:
    reference: str
    buyer_name: str
    buyer_email: str
    payment_method: str
    subtotal: Decimal
    fee: Decimal
    total: Decimal
    issued_at_utc: datetime
    message: str

File: Baasket.LogisticsV2/CheckoutReceipt.cs
namespace Baasket.LogisticsV2;

/// <summary>
/// Immutable receipt produced after payment confirmation.
/// Maps to the existing Python <c>PaymentReceipt</c> concept.
/// </summary>
public sealed record CheckoutReceipt(
    string Reference,
    string BuyerName,
    string BuyerEmail,
    string PaymentMethod,
    decimal Subtotal,
    decimal Fee,
    decimal Total,
    DateTime IssuedAtUtc,
    string Message);
